### PR TITLE
[FW][FIX] account (fixed forward port to v15)

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Adapted https://github.com/odoo/odoo/pull/98498 to https://github.com/odoo/odoo/commit/c0f879f6b831a49f29317d3841d88dbb097b2ace.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/tests/test_pos_basic_config.py [hunk 1]
```diff
@@ -318,6 +318,68 @@ def test_orders_with_zero_valued_invoiced(self):
...
+                },
+                'cash_statement': [
+                    ((100, ), {
+                        'line_ids': [
+                            {'account_id': self.cash_pm1.outstanding_account_id.id, 'partner_id': False, 'debit': 100, 'credit': 0, 'reconciled': False},
+                            {'account_id': self.cash_pm1.receivable_account_id.id, 'partner_id': False, 'debit': 0, 'credit': 100, 'reconciled': True},
+                        ]
+                    }),
+                ],
+                'bank_payments': [],
```
Comment (MiquelRForgeFlow) [inline addons/point_of_sale/tests/test_pos_basic_config.py:379]: not sure about this part, but test doesn't fail

Comment (MiquelRForgeFlow) [reply addons/point_of_sale/tests/test_pos_basic_config.py:379]: @caburj please, as you did the refactor of the pos tests (https://github.com/odoo/odoo/commit/c0f879f6b831a49f29317d3841d88dbb097b2ace), could you check if this part is correct? this test comes from https://github.com/odoo/odoo/pull/98498/commits/99030bdb17489bb8e18bf9bcc867f551368f77b7

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Comment by MiquelRForgeFlow** [comment]
@william-andre ?

**Comment by william-andre** [comment]
@robodoo rebase-ff

**Comment by MiquelRForgeFlow** [comment]
Selection_519

This test `test_import_fec_demo_file` from `l10n_fr_fec_import` is a bit ugly, it should be written different :S

What do you recommend? I could replace the 256 by 100, but ...

**Comment by MiquelRForgeFlow** [comment]
@william-andre I investigated the results of that test and compared them with the fix and without the fix. The data is as follows:

- Before the fix:
  - 100 lines with reconciled = TRUE and have full_reconcile_id
  - 156 lines with reconciled = FALSE and have full_reconcile_id
->  thus making a total of 256

- After the fix:
  - 100 lines with reconciled = TRUE and have full_reconcile_id
  - 0 lines with reconciled = FALSE and have full_reconcile_id
->  thus making a total of 100

Obviously if a line is not reconciled, it shouldn't have full_reconcile_id. Thus the fixed versions is better. Those 156 unreconciled lines that had a full_reconcile_id now with the fix don't have a full_reconcile_id.

I will make a PR in enterprise to fix that test.

**Comment by lordkrandel** [comment]
> Selection_519
> 
> This test `test_import_fec_demo_file` from `l10n_fr_fec_import` is a bit ugly, it should be written differently :S

It was just a fast test to just see that things get reconciled.
I get that checking the raw number of reconciled lines is not the best, but there was no test at all before, so now at least we notice when something breaks - which is exactly the case. 
If the number has changed, then fixing it to 100 now is correct imho.

**Comment by MiquelRForgeFlow** [comment]
Enterprise PR https://github.com/odoo/enterprise/pull/30930.

**Comment by william-andre** [comment]
@robodoo retry

**Pull request closed without merging**